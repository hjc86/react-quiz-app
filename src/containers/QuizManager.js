import React, { Component } from 'react';
import Quiz from '../components/Quiz';
import Score from '../components/Score';
// import styles from './Q'

class QuizManager extends Component {
    constructor(props){
        super(props)
        
        this.state={
            questionsAnswered: 0,
            playerIndex: 0,
            finished: false,
            quizQA: null,
            score: 0,
            scoreArray: [],
            carryOn: true, 
            winner: null,
            quizFinished: false

        }  
    }
    
    updateAnsweredQuestions=(questionsAnswered)=>{
        this.setState({
            questionsAnswered: questionsAnswered
        })

        if(this.state.questionsAnswered===5 && this.state.playerIndex<this.props.playerInfo.userName.length-1){
            this.setState({
                playerIndex: this.state.playerIndex+1,
                questionsAnswered: 0,
                
            })
        }

        // if(this.state.questionsAnswered===5 && this.state.playerIndex<this.props.playerInfo.userName.length-1){
    
        // }
    }
    
    updateScore= (playerObj)=>{
        if(this.state.questionsAnswered === 4){
            this.setState({
                scoreArray: [...this.state.scoreArray, playerObj.score],
                score: playerObj.score   
             })
        }

        if(this.state.scoreArray.length===this.props.playerInfo.userName.length){
            this.setState({
                carryOn: false
            })
        } 
    } 
    
    render() {
        return (
            <div>
                        {this.state.questionsAnswered === 5?
                        
                            <Score scoreArray={this.state.scoreArray} userName={this.props.playerInfo.userName} /> :
                
                            <Quiz
                            playerInfo={this.props.playerInfo} 
                            currentPlayer={this.props.playerInfo.userName[this.state.playerIndex]} 
                            answeredQuestions={this.updateAnsweredQuestions}
                            score={this.updateScore}/>
                    
                        }

            </div>
        
        )
    }
}


export default QuizManager