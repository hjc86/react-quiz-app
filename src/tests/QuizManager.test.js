import React from 'react';
import ReactDOM from 'react-dom'
import {shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import QuizManager from '../containers/QuizManager'

describe(QuizManager, () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<QuizManager />));
    
    // beforeEach(() => wrapper.setProps({playerInfo: {userName: 'Tom'}}))
    beforeEach(() => wrapper.setState({questionsAnswered: 5}))
    
    //check rendering and mounting of <QuizManager /> component //
    it('should mount to DOM', () => {
        const div = document.createElement('div');
        ReactDOM.render(<QuizManager />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(<QuizManager />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    
   
    
   
    
    // it('should render a single div <div />', () => {
    //     expect(wrapper.find('div').length).toEqual(1);
    // });

   
    // it('should render two <p />', () => {
    //     let wrapper =shallow(<QuizManager />)
    //     expect(wrapper.find('p').length).toEqual(1);
    //  });
 
    //test conditional rendering //

    //beforeEach(()=>wrapper.setState({QuestionsAnswered: 5}))

})