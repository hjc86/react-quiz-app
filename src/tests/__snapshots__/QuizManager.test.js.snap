// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizManager/ > renders correctly 1`] = `
<div>
  waiting for Ts round to load load
</div>
`;
